pkg/scheduler/cache/cache.go

```diff
@@ -17,43 +19,38 @@ type Cache interface {
 }
 
 type schedulerCache struct {
-	mutex    sync.RWMutex
-	clusters map[string]*v1alpha1.Cluster
+	clusterLister lister.ClusterLister
 }
 
 // NewCache instantiates a cache used only by scheduler.
-func NewCache() Cache {
+func NewCache(clusterLister lister.ClusterLister) Cache {
 	return &schedulerCache{
-		clusters: make(map[string]*v1alpha1.Cluster),
+		clusterLister: clusterLister,
 	}
 }
 
+// AddCluster does nothing since clusterLister would synchronize automatically
 func (c *schedulerCache) AddCluster(cluster *v1alpha1.Cluster) {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-	c.clusters[cluster.Name] = cluster
 }
 
+// UpdateCluster does nothing since clusterLister would synchronize automatically
 func (c *schedulerCache) UpdateCluster(cluster *v1alpha1.Cluster) {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-	c.clusters[cluster.Name] = cluster
 }
 
+// DeleteCluster does nothing since clusterLister would synchronize automatically
 func (c *schedulerCache) DeleteCluster(cluster *v1alpha1.Cluster) {
```

Comment:
Is there a specific reason for keeping these empty methods(`AddCluster`, `UpdateCluster`, `DeleteCluster`)?

Reply:
Thanks for your review, as I mentioned in #727

> Besides, IMHO AddCluster,UpdateCluster and DeleteCluster are not necessary any more because the indexer would synchronize automatically, shall we delete these methods in `Cache` interface?

The reason for keeping these methods is to implement `Cache` interface, but it seems that these methods are not necessary any more, shall I delete them in `Cache` interface?

Reply:
Once i added the cache interface, i meant  maybe in the future we need extend the function there. Maybe we want to convert the original cluster object to ClusterInfo directly.

Reply:
So, let's clean up this interface. And we can add them back if there is a need in the future.

@dddddai